### PR TITLE
[WJ-462] Implement PHP integration for ftml FFI

### PR DESCRIPTION
**Pull request by emmiegit**
The class for handling parsing/rendering in Wikidot is `WikiTransformation` (imo a misleading name). To get us closer to having separate, configurable Text\_Wiki vs ftml parser/renderer backends, I have moved it to a new module at `web/Wikitext`. There is a subdirectory `web/Wikitext/FTML` for all ftml-related classes and implementations.

Other than the above move, this PR does *not* change the native Wikidot parser/renderer in any way. It simply introduces the necessary code to render using ftml, which is unused.

It also fixes an issue with the Rust side where the meta list wasn't freed in the `ftml_destroy_html_output()` function. The PhpStorm IDE also made some stylistic modifications.

The integration was tested locally and confirmed to work:

----

**HTML render test:**
HTML render test

**Text render test:** (in pre)
Text render test

### CODE REVIEW

**Review by Monkatraz — APPROVED** [review]
I'm not very qualified to review this, but in principle I don't see anything wrong.

well, except for the sections of the diff where it reveals just little bits of old wikidot code. true horror

File: web/php/Wikitext/FTML/FtmlBackend.php [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<?php
```
Comment (danieltharp) [inline web/php/Wikitext/FTML/FtmlBackend.php]: The Wikidot namespace is intended for legacy code. Any new code should go in one of the laravel-provided folders, likely the `app` folder as everything there inherits the `Wikijump` namespace.

**Review by danieltharp — CHANGES_REQUESTED** [review]
If you move the new FTML stuff out of `web/php` and into `app` and renamespace it to `Wikijump` I think this will be fine.

File: web/php/Wikitext/FTML/FtmlBackend.php (hunk 1 above)
Comment (emmiegit) [reply web/php/Wikitext/FTML/FtmlBackend.php]: Ah alright, I will adjust accordingly.

File: CODEOWNERS [hunk 2]
```diff
@@ -5,6 +5,7 @@
 # The ftml (Foundation Text Markup Language) service provides
 # parsing and rendering of wikitext.
 ftml                                @ammongit
+web/php/Wikitext                    @ammongit
```
Comment (rossjrw) [inline CODEOWNERS]:
```suggestion
web/app/Services/Wikitext           @ammongit
```

File: install/aws/dev/docker/php-fpm/etc/php/99-ffi.ini [hunk 3]
```diff
@@ -0,0 +1 @@
+ffi.enable=true
```
Comment (rossjrw) [inline install/aws/dev/docker/php-fpm/etc/php/99-ffi.ini:1]: What does the 99 in the filename mean?

**Review by rossjrw — APPROVED** [review]

File: install/aws/dev/docker/php-fpm/etc/php/99-ffi.ini (hunk 3 above)
Comment (emmiegit) [reply install/aws/dev/docker/php-fpm/etc/php/99-ffi.ini:1]: [It's a UNIX convention](https://unix.stackexchange.com/questions/134520/what-is-the-number-prefix-in-config-files-from-d-directory). We probably don't need numbering but I'm doing it for consistency.

**Review by danieltharp — APPROVED** [review]

**Comment by emmiegit** [comment]
thanks @pxdnbluesoul @Monkatraz @rossjrw

**Pull request merged**